### PR TITLE
Fixes #7581: Remove unapplying patches

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7581

### CODE REVIEW

File: rudder-agent/SOURCES/patches/cfengine/0018-do-not-remove-cflags-when-compiling.patch [hunk 1]
```diff
@@ -1,10 +0,0 @@
---- a/configure.ac	2015-10-12 11:48:53.451469571 +0200
-+++ b/configure.ac	2015-10-12 11:50:13.759467288 +0200
-@@ -54,7 +54,6 @@
- dnl It is restored later.
- dnl
- ENV_CFLAGS="$CFLAGS"
--CFLAGS=
```
Comment (jooooooon) [inline rudder-agent/SOURCES/patches/cfengine/0018-do-not-remove-cflags-when-compiling.patch]: This line is no longer in CFEngine 3.6.3's configure.ac and I don't see any other lines in there that reset CFLAGS, so I assume this patch is no longer necessary.

File: rudder-agent/SOURCES/patches/cfengine/07-define-classes-when-package-is-up-to-date.patch [hunk 2]
```diff
@@ -1,13 +0,0 @@
-diff -ur a/cf-agent/verify_packages.c b/cf-agent/verify_packages.c
```
Comment (jooooooon) [inline rudder-agent/SOURCES/patches/cfengine/07-define-classes-when-package-is-up-to-date.patch]: This is a duplicate patch with 0017-define-classes-when-package-is-up-to-date.patch. They do the same thing, therefore the second doesn't apply, causing errors.

**Comment by jooooooon** [comment]
Don't merge yet

**Comment by jooooooon** [comment]
In 3.2 branch the *other* patch for define-classes-when-package-is-up-to-date was removed, so we should remove the same one to avoid the merge ending up with neither

**Comment by jooooooon** [comment]
PR updated

File: rudder-agent/SOURCES/patches/cfengine/0017-define-classes-when-package-is-up-to-date.patch [hunk 3]
```diff
@@ -1,13 +0,0 @@
-diff -ur a/cf-agent/verify_packages.c b/cf-agent/verify_packages.c
```
Comment (jooooooon) [inline rudder-agent/SOURCES/patches/cfengine/0017-define-classes-when-package-is-up-to-date.patch]: This patch is a duplicate of patch 07-define-classes-when-package-is-up-to-date.patch and has already been removed in the 3.2 branch.

**Comment by jooooooon** [comment]
OK, good to go again.

**Comment by peckpeck** [comment]
OK merging

**Pull request merged**